### ISSUE TITLE
model: Impelment a way to pass model specific config options

### ISSUE THREAD

**Issue by pdxjohnny** [intel/dffml#22]
For `tensorflow/DNN` DFFML Model, `hidden_units` is hardcoded

https://github.com/intel/dffml/blob/ba31ae715f0ff219ca36b99b5a912e598427a8c2/model/tensorflow/dffml_model_tensorflow/model/dnn.py#L192

This sucks, there needs to be away for models to define what arguments they take, and then have those passed to the function instantiating them.

For instance, on the Iris dataset, changing to `[12, 40, 15]` gets us ~96% accuracy in ~1201 steps.

**Comment by sudharsana-kjl** [comment]
Hi! I have a proposed idea for this. Could you tell me if this is fine?

Considering that the value of ```hidden_units``` might change according to the model and dataset we choose

- Can we have a command line flag ```-ModelParams``` which will take the variable name to be changed and its corresponding value?

Something like this:
```
dffml train \
  -model dnn \
  -sources csv=iris_training.csv \
  -classifications 0 1 2 \
  -features \
    def:SepalLength:float:1 \
    def:SepalWidth:float:1 \
    def:PetalLength:float:1 \
    def:PetalWidth:float:1 \
  -num_epochs 3000 \
  -steps 20000 \
  -log debug
  -ModelParams \
     hidden_units=[12,40,15]
```

This flag could be optional. If no such flag is specified, then it'll choose the default value hardcoded in the model

**Comment by pdxjohnny** [comment]
Awesome! Right now DFFML only uses CLI flags, no config files, that'll change eventually. So for know if you want to work on this go with solution 1.

Check out https://docs.python.org/3.6/library/argparse.html and https://github.com/intel/dffml/blob/master/dffml/util/cli.py

dffml.util.cli will probably be where most of the changes for this end up. (Argparse is weird, this may take a while to do).

Also, there's going to be a lot of back and forth in the PR comments on whatever gets done for this, so make the titile prefaced with `WIP: ` then remove the WIP (work in progresss) when you think it's ready to be merged.

I'm trying to keep things rather particular, so know that I'm not picking on you if theres a lot of changes requested.
